### DISCUSSION TITLE
RFC: Changing the License

### DISCUSSION THREAD

**Discussion by eudoxia0** [austral/austral#419] [Ideas]
The compiler's license is GPLv3. I intended to apply this just to the compiler, naturally. But I never got around to specifying licenses for the runtime/builtins, and also, I never got around to adding something like GCC's runtime linking exception.

I've opened a few issues to relicense the runtime, builtins, and standard library (#390, #409, #410). However, I'm not sure if this is sufficient to comply with the wording of the GPL.

GCC is GPL, and has a "Runtime Linking Exception" so that compiled code is not GPL. However, the text is very specific to the C model of compilation and linking. Writing a linking exception for this compiler would require adjusting the legalese to fit Austral's compilation model, which requires legal knowledge I don't have, and it probably would be specific to the choice of backend+compilation model.

Needless to say: this has all been a _huge_ headache.

My concern is that people will be put off either by the GPL in general or by the licensing ambiguity around compiler output.

I think the wisest course of action would be to relicense the whole repository outright, and leave nothing under the GPL. I think Apache 2.0 is a good choice, since it's permissive and well-understood.

I think it's tractable to get permission from every contributor, since so far only eight other people have contributed to the repository.

**Comment by ckp95** [comment]
seems sensible, copyright law is complicated and seems best to avoid ambiguity.

**Comment by eudoxia0** [comment]
Fixed by #435
